src/schema.lua

```diff
@@ -23,10 +43,40 @@ return {
             type = "record",
             fields = {
                 {model_path = {required = true, type = "string"}},
-                {policy_path = {required = true, type = "string"}},
-                {username = {required = true, type = "string"}}
-            }
+                {policy_path = {type = "string"}},
+                {username = {required = true, type = "string"}},
+                {adapter = {type = "string", one_of = adapter_type_arr, default = "file"}},
+                {db_info = {
+                    type = "record",
+                    fields = {
+                        {db_type = {type = "string", one_of = db_type_arr}},
+                        {database = {type = "string"}},
+                        {username = {type = "string"}},
+                        {password = {type = "string"}},
+                        {host = {type = "string"}},
+                        {port = {type = "string"}}
+                    }
+                }}
+            },
+            custom_validator = function (config)
+                local adapter_type = config.adapter
+                if adapter_type == "file" then
+                    if isNotEmptyStr(config.policy_path) then
+                        return true
+                    end
+                    return false, "default use FileAdapter, you should specify model_path and policy_path"
+                end
+                local db_type = config.db_info.db_type
+                if not isNotEmptyStr(db_type) then
+                    return false, "you are using sql-adapter, please specify db_type and other required db info"
+                end
+                if adapter_type == "luasql" and (not contains(db_type, luasql_arr)) then
+                    return false, db_type.."not supported when adapter_type is" .. adapter_type
```

Comment:
```suggestion
                    return false, db_type.." not supported when adapter_type is " .. adapter_type
```
By spaces I meant before and after the string.